### PR TITLE
Avoid cloning elements in forms

### PR DESCRIPTION
**Pull request by djhi**
After this PR, I'll make a few others to ensure that every inputs handle the `fullWidth` property correctly. I'll also add stories for each input.

### CODE REVIEW

File: packages/ra-ui-materialui/src/form/FormTab.tsx [hunk 1]
```diff
@@ -102,3 +77,13 @@ export interface FormTabProps {
 FormTab.displayName = 'FormTab';
 
 const hiddenStyle = { display: 'none' };
+
+const DefaultComponent = ({ children, ...props }) => (
```
[the remark this answers is not in the document]
Comment (djhi) [reply packages/ra-ui-materialui/src/form/FormTab.tsx:81]: I don't get your comment

File: packages/ra-ui-materialui/src/form/SimpleFormView.tsx [hunk 2]
```diff
@@ -69,18 +44,25 @@ SimpleFormView.propTypes = {
     validate: PropTypes.func,
 };
 
+const DefaultComponent = ({ children, ...props }) => (
+    <Stack
+        alignItems="flex-start"
+        sx={{ paddingLeft: 1, paddingRight: 1 }}
+        {...props}
+    >
+        {children}
```
[the remark this answers is not in the document]
Comment (djhi) [reply packages/ra-ui-materialui/src/form/SimpleFormView.tsx:53]: That's why this PR is a **draft**

File: UPGRADE.md [hunk 3]
```diff
@@ -1413,6 +1413,85 @@ const PostShow = () => (
 };
 ```
 
+## No More Props Injection In `<SimpleForm>` and `<TabbedForm>`
```
[the remark this answers is not in the document]
Comment (djhi) [reply UPGRADE.md:1416]: They is h2 section and two h3 sub sections because the fact that we don't inject props implies two changes

File: docs/CreateEdit.md [hunk 4]
```diff
@@ -818,30 +818,34 @@ export const PostCreate = () => (
 
 ### Label Decoration
 
-`<SimpleForm>` scans its children for the `addLabel` prop, and automatically wraps a child in a `<Labeled>` component when found. This displays a label on top of the child, based on the `label` prop. This is not necessary for `<Input>` components, as they already contain their label. Also, all the react-admin `<Field>` components have a default prop `addLabel: true`, which explains why react-admin shows a label on top of Fields when they are used as children of `<SimpleForm>`. 
+All react-admin inputs handle the display of their label by wrapping their content inside a `<Labeled>` component.
```
[the remark this answers is not in the document]
Comment (djhi) [reply docs/CreateEdit.md:821]: It is actually true for many inputs (`ArrayInput`, CheckboxGroupInput`, `FileInput`, `RadioButtonGroupInput`, `SelectArrayInput` and `SelectInput`). The others uses MUI default label

File: docs/CreateEdit.md [hunk 5]
```diff
@@ -969,31 +973,37 @@ export const PostEdit = () => (
 
 ### Label Decoration
 
-`<FormTab>` scans its children for the `addLabel` prop, and automatically wraps a child in a `<Labeled>` component when found. This displays a label on top of the child, based on the `label` prop. This is not necessary for `<Input>` components, as they already contain their label. Also, all the react-admin `<Field>` components have a default prop `addLabel: true`, which explains why react-admin shows a label on top of Fields when they are used as children of `<FormTab>`. 
+All react-admin inputs handle the display of their label by wrapping their content inside a `<Labeled>` component.
```
[the remark this answers is not in the document]
Comment (djhi) [reply docs/CreateEdit.md:976]: We have this section inside the `SimpleForm` and the `TabbedForm` sections

File: examples/simple/src/comments/CommentCreate.tsx [hunk 6]
```diff
@@ -23,13 +23,9 @@ const CommentCreate = () => (
...
                 sort={defaultSort}
             />
-            <TextInput
-                source="author.name"
-                validate={minLength(10)}
-                fullWidth
-            />
+            <TextInput source="author.name" validate={minLength(10)} />
             <DateInput source="created_at" defaultValue={now} />
-            <TextInput source="body" fullWidth={true} multiline={true} />
```
[the remark this answers is not in the document]
Comment (djhi) [reply examples/simple/src/comments/CommentCreate.tsx:32]: We actually don't need that for many inputs (MUI ones that already support `fullWidth`). I'll have to ensure all our inputs also support `fullWidth`

File: docs/TabbedShowLayout.md [hunk 7]
```diff
@@ -28,7 +28,7 @@ export const PostShow = () => (
                 <TextField source="teaser" />
             </Tab>
             <Tab label="body" path="body">
-                <RichTextField source="body" addLabel={false} />
+                <RichTextField source="body" label={false} />
```
[the remark this answers is not in the document]
Comment (djhi) [reply docs/TabbedShowLayout.md:31]: It's already mentioned in the layouts documentation but I added it there as well

File: docs/TabbedShowLayout.md [hunk 8]
```diff
@@ -38,7 +38,7 @@ export const PostShow = () => (
                 <TextField label="Nb views" source="views" />
             </Tab>
             <Tab label="comments" path="comments">
-                <ReferenceManyField reference="comments" target="post_id" addLabel={false}>
```
[the remark this answers is not in the document]
Comment (djhi) [reply docs/TabbedShowLayout.md:41]: It is

File: docs/Labeled.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+     </Car>
+  </Show>
+);
+```
+
+## API
+
+* [`Labeled`]
+
+[`Labeled`]: https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/detail/Labeled.tsx
```
Comment (WiXSL) [inline docs/Labeled.md:79]: This link doesn't exist

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/input/Labeled.tsx [hunk 10]
```diff
@@ -44,40 +42,32 @@ export const Labeled = (props: LabeledProps) => {
...
+                >
+                    <FieldTitle
+                        label={label}
+                        source={source}
+                        resource={resource}
+                        isRequired={isRequired}
+                    />
+                </InputLabel>
+            )}
+            <div className={LabeledClasses.value}>{children}</div>
```
[the remark this answers is not in the document]
Comment (djhi) [reply packages/ra-ui-materialui/src/input/Labeled.tsx]: They have been merged

File: packages/ra-ui-materialui/src/form/FormTab.tsx [hunk 11]
```diff
@@ -44,48 +44,23 @@ export const FormTab = (props: FormTabProps) => {
...
-                                input={input}
-                                record={record}
-                                resource={resource}
-                                variant={input.props.variant || variant}
-                                margin={input.props.margin || margin}
-                            />
-                        )
-                )}
-            </span>
+                {children}
```
[the remark this answers is not in the document]
Comment (djhi) [reply packages/ra-ui-materialui/src/form/FormTab.tsx:48]: Im not sure I get the question.

We don't automatically decorate inputs nor fields in forms anymore. Inputs are responsible for their labels and you must wrap fields in `<Labeled>`

File: packages/ra-ui-materialui/src/form/TabbedForm.spec.tsx [hunk 12]
```diff
@@ -182,30 +164,33 @@ describe('<TabbedForm />', () => {
         ).toEqual(false);
     });
 
-    it('should sync tabs with location by default', () => {
+    it.only('should sync tabs with location by default', () => {
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/form/TabbedForm.spec.tsx:167]: `only` spotted !

File: packages/ra-ui-materialui/src/form/TabbedForm.spec.tsx [hunk 13]
```diff
@@ -182,30 +164,33 @@ describe('<TabbedForm />', () => {
...
+                                source="description"
+                                validate={minLength(10)}
+                            />
+                        </FormTab>
+                    </TabbedForm>
+                </ResourceContextProvider>
             </AdminContext>
         );
 
+        screen.debug();
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/form/TabbedForm.spec.tsx:193]: `debug` spotted

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: UPGRADE.md [hunk 14]
```diff
@@ -1454,6 +1454,91 @@ export const PostEdit = () => (
...
+-        <SimpleForm margin="normal" variant="outlined">
++        <SimpleForm>
+-            <TextInput source="title" />
++            <TextInput source="title" margin="normal" variant="outlined" />
+        </SimpleForm>
+    </Create>
+)
+```
+
+You have several options when leveraging the [theme component overrides](https://mui.com/customization/theme-components/):
```
Comment (WiXSL) [inline UPGRADE.md:1504]: In line 1490 you already mention this link.
Maybe this entire line could be omitted

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: UPGRADE.md [hunk 15]
```diff
@@ -1454,6 +1454,91 @@ export const PostEdit = () => (
 
 Use the `<Title>` component instead.
 
+## No More Props Injection In `<SimpleForm>` and `<TabbedForm>`
+
+The form components used to clone their children, inspect their props to handle labels and inject the `margin` and `variant` props.
+
+This is no longer the case and makes custom layout easier.
+
+### `<SimpleForm>` and `<TabbedForm>` No Longer Clone Their Children
```
Comment (WiXSL) [inline UPGRADE.md:1463]: Once again, this is wonderful! 💪

File: docs/Fields.md [hunk 16]
```diff
@@ -477,43 +466,22 @@ export default TextField;
 
 ## Adding A Label To Custom Field Components
 
-When you use one of the react-admin `Field` components in an `Edit`, `Create` or `Show` view, react-admin includes a label on top of the field value, as in the following example:
-
-![field labels](./img/field-addlabel.png)
-
-For your custom fields, however, the label doesn't appear by default. You need to opt-in to this feature by setting the `addLabel` prop to `true` in the `defaultProps`.
+If you want to have a label for your custom fields in an `Edit`, `Create` or `Show` view, uou need to wrap your component inside a `<Labeled>` component:
```
Comment (WiXSL) [inline docs/Fields.md:469]:
```suggestion
If you want to have a label for your custom fields in an `Edit`, `Create` or `Show` view, you need to wrap your component inside a `<Labeled>` component:
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/Labeled.md [hunk 17]
```diff
@@ -0,0 +1,79 @@
+---
+layout: default
+title: "Labeled"
+---
+
+# `<Labeled>`
```
Comment (WiXSL) [inline docs/Labeled.md:6]: Could you update `Reference.md` reference of Labeled component to point to this new file?

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/Labeled.md [hunk 18]
```diff
@@ -0,0 +1,79 @@
...
+
+`<Labeled>` adds a label on top of its child if the child exposes a `label` prop, or if it's a field with a `source` prop. It's used in Show layouts, to display both a field and its label.
```
Comment (WiXSL) [inline docs/Labeled.md:8]: `. It's used in Show layouts...`
This explanation could be improved (?), since it is used in other layouts as well

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/SimpleShowLayout.md [hunk 19]
```diff
@@ -212,9 +215,9 @@ To override the style of all instances of `<SimpleShowLayout>` using the [materi
 ## API
 
 * [`<SimpleShowLayout>`]
-* [`<FieldWithLabel>`]
+* [`<Labeled>`]
 * [`useRecordContext`]
 
 [`<SimpleShowLayout>`]: https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/detail/SimpleShowLayout.tsx
-[`<FieldWithLabel>`]: https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/detail/FieldWithLabel.tsx
+[`<Labeled>`]: https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/detail/Labeled.tsx
```
Comment (WiXSL) [inline docs/SimpleShowLayout.md:222]:
```suggestion
[`<Labeled>`]: https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/Labeled.tsx
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/TabbedShowLayout.md [hunk 20]
```diff
@@ -294,10 +324,10 @@ To style the tabs, the `<Tab>` component accepts two props:
 
 * [`<TabbedShowLayout>`]
 * [`<Tab>`]
-* [`<FieldWithLabel>`]
+* [`<Labeled>`]
 * [`useRecordContext`]
 
-[`<FieldWithLabel>`]: https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/detail/FieldWithLabel.tsx
+[`<Labeled>`]: https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/detail/Labeled.tsx
```
Comment (WiXSL) [inline docs/TabbedShowLayout.md:330]:
```suggestion
[`<Labeled>`]: https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/Labeled.tsx
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/Labeled.tsx [hunk 21]
```diff
@@ -44,12 +50,15 @@ export const FieldWithLabel = ({
         <div className={className}>{children}</div>
     );
 
-FieldWithLabel.displayName = 'FieldWithLabel';
+Labeled.displayName = 'Labeled';
 
 export interface FieldWithLabelProps {
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/Labeled.tsx:55]:
```suggestion
export interface LabeledProps {
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/Labeled.tsx [hunk 22]
```diff
@@ -6,36 +6,42 @@ import { styled } from '@mui/material/styles';
...
-export const FieldWithLabel = ({
-    label,
+export const Labeled = ({
     children,
     className = '',
+    isRequired,
+    label,
+    resource,
+    source,
 }: FieldWithLabelProps) =>
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/Labeled.tsx:27]:
```suggestion
}: LabeledProps) =>
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/input/CheckboxGroupInput.tsx [hunk 23]
```diff
@@ -170,9 +171,10 @@ export const CheckboxGroupInput: FunctionComponent<CheckboxGroupInputProps> = pr
                 label={label}
                 source={source}
                 resource={resource}
-                className={className}
+                className={clsx('ra-input', `ra-input-${source}`, className)}
                 isRequired={isRequired}
                 margin={margin}
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/input/CheckboxGroupInput.tsx:176]: I'm getting a TypeScript error here, `margin` is not recognized

File: packages/ra-ui-materialui/src/form/FormInput.tsx [hunk 24]
```diff
@@ -5,7 +5,7 @@ import PropTypes from 'prop-types';
 import clsx from 'clsx';
 import { RaRecord } from 'ra-core';
 
-import { Labeled } from '../input/Labeled';
+import { Labeled } from '../Labeled';
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/form/FormInput.tsx:8]: I'm getting a TypeScript error here, `id` is not recognized in Labeled

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/form/FormInput.tsx [hunk 25]
```diff
@@ -5,7 +5,7 @@ import PropTypes from 'prop-types';
...
 
 export const FormInput = <
     RecordType extends RaRecord | Omit<RaRecord, 'id'> = any
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/form/FormInput.tsx:11]: FormInput is still asking for `addLabel`

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/Labeled.md (hunk 18 above)
Comment (djhi) [reply docs/Labeled.md:8]: Show layouts with an **s** ;-)

File: UPGRADE.md (hunk 14 above)
Comment (djhi) [reply UPGRADE.md:1504]: I don't mind having links where they are needed :man_shrugging:

File: docs/Labeled.md (hunk 17 above)
Comment (djhi) [reply docs/Labeled.md:6]: Done

File: docs/Reference.md [hunk 26]
```diff
@@ -70,7 +70,7 @@ title: "Reference"
 * [`<ImageField>`](./ImageField.md)
 * [`<ImageInput>`](./ImageInput.md)
 * [`<ImageInputPreview>`](./ImageInput.md#imageinput)
-* [`<Labeled>`](./Inputs.md#using-labeled)
```
Comment (WiXSL) [inline docs/Reference.md:73]: In `Fields.md` there is one more reference to `Inputs.md#using-labeled` that could be changed for `Labeled.md`

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/Reference.md (hunk 26 above)
Comment (djhi) [reply docs/Reference.md:73]: Can't find any

Comment (WiXSL) [reply docs/Reference.md:73]: Right, you removed that part in your `Fields.md`

File: docs/Fields.md [hunk 27]
```diff
@@ -477,43 +466,22 @@ export default TextField;
 
 ## Adding A Label To Custom Field Components
 
-When you use one of the react-admin `Field` components in an `Edit`, `Create` or `Show` view, react-admin includes a label on top of the field value, as in the following example:
-
-![field labels](./img/field-addlabel.png)
-
-For your custom fields, however, the label doesn't appear by default. You need to opt-in to this feature by setting the `addLabel` prop to `true` in the `defaultProps`.
+If you want to have a label for your custom fields in an `Edit`, `Create` or `Show` view, you need to wrap your component inside a `<Labeled>` component:
```
Comment (fzaninotto) [inline docs/Fields.md:469]: No in a Show view. Besides, I think this tip is misleading. It's not the responsibility of the field anymore to add a label.

File: packages/ra-ui-materialui/src/Labeled.tsx [hunk 28]
```diff
@@ -0,0 +1,105 @@
...
+    component?: ElementType;
+    fullWidth?: boolean;
+    htmlFor?: string;
+    isRequired?: boolean;
+    label?: string | ReactElement | false;
+    resource?: string;
+    source?: string;
+}
+
+const PREFIX = 'RaFieldWithLabel';
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/Labeled.tsx:83]:
```suggestion
const PREFIX = 'RaLabeled';
```

File: packages/ra-ui-materialui/src/defaultTheme.ts [hunk 29]
```diff
@@ -65,6 +65,26 @@ export const defaultTheme = {
                 },
             },
         },
+        MuiFormControl: {
+            styleOverrides: {
+                root: {
+                    width: 256,
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/defaultTheme.ts:71]: Not sure it renders correctly on Mobile

File: packages/ra-ui-materialui/src/form/FormTab.tsx [hunk 30]
```diff
@@ -35,7 +34,9 @@ export const FormTab = (props: FormTabProps) => {
 
     const renderContent = () => (
         <FormGroupContextProvider name={value.toString()}>
-            <span
+            <Stack
+                alignItems="flex-start"
+                sx={{ paddingLeft: 1, paddingRight: 1 }}
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/form/FormTab.tsx:39]: Can users override this one? We usually use a styled root component with classes for the inner components

File: packages/ra-ui-materialui/src/form/FormTab.tsx [hunk 31]
```diff
@@ -44,44 +45,14 @@ export const FormTab = (props: FormTabProps) => {
...
-            </span>
+                {children}
+            </Stack>
         </FormGroupContextProvider>
     );
 
     return intent === 'header' ? renderHeader() : renderContent();
 };
 
-FormTab.propTypes = {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/form/FormTab.tsx:67]: Why did you remove this one?

File: packages/ra-ui-materialui/src/input/AutocompleteInput.tsx [hunk 32]
```diff
@@ -410,7 +409,7 @@ If you provided a React element for the optionText prop, you must also provide t
     };
 
     return (
-        <Root className={className}>
```
[the remark this answers is not in the document]
Comment (fzaninotto) [reply packages/ra-ui-materialui/src/input/AutocompleteInput.tsx:415]: So the class name works, but I don't understand the rationale of the change as it doesn't allow to override the styles of the createElement anymore

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/defaultTheme.ts (hunk 29 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/defaultTheme.ts:71]: True, we didn't do better before though

File: packages/ra-ui-materialui/src/form/FormTab.tsx (hunk 30 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/form/FormTab.tsx:39]: `.RaTabbedForm .MuiStack`

File: packages/ra-ui-materialui/src/input/AutocompleteInput.tsx (hunk 32 above)
[the remark this answers is not in the document]
Comment (djhi) [reply packages/ra-ui-materialui/src/input/AutocompleteInput.tsx:415]: Well the createElement is your own component so you can already style it how you want

**Review by fzaninotto — APPROVED** [review]
